### ISSUE TITLE
Invalid `.eb` file saved upon successful build of a `.yeb` file

### ISSUE THREAD

**Issue by riccardomurri** [hpcugent/easybuild-framework#1695]
I have tried to build LinBox 1.4.0 using a `.yeb` file; the build
succeeded, but then the file that was saved into my `ebfiles_repo`
directory is an incorrect mixture of YAML and Python syntax.

It seems that EB currently assumes that an input file has Python
syntax and applies some text transform to it, which generates garbage
when the input file is YAML instead.

For reference, here's the file that EB output; the original input file
is the part at lines 2 through 52::

```
    # Built with EasyBuild version 2.7.0 on 2016-03-24_17-24-34
    %YAML 1.2
    ---

    ##
    # This file is an EasyBuild recipe; see https://github.com/hpcugent/easybuild
    #
    # Copyright:: Copyright (c) 2016 Riccardo Murri <riccardo.murri@gmail.com>
    # Authors::   Riccardo Murri <riccardo.murri@gmail.com>
    # License::   GPL
    #
    ##

    name: LinBox
    version: 1.4.0
    description:
      C++ library for exact, high-performance linear algebra
    homepage: 'http://linalg.org/'
    moduleclass: math

    source_urls:
      - 'https://github.com/linbox-team/linbox/archive'
    sources:
      - 'v1.4.0.zip'

    easyblock: ConfigureMake
    toolchain:
      name: foss
      version: 2015a
    builddependencies:
      - [Autoconf, 2.69]
      - [Automake, 1.15]
      - [libtool, 2.4.6]
    dependencies:
      - FFLAS-FFPACK: 2.2.0
      - Givaro: 4.0.1
    # configure with: autoreconf -i && ./configure
    preconfigopts: 'env NOCONFIGURE=1 ./autogen.sh && '
    configopts: '--with-givaro=$EBROOTGIVARO --with-fflas-ffpack=$EBROOTFFLASMINFFPACK --enable-openmp'

    sanity_check_paths:
      files: ['bin/linbox-config', 'include/linbox/linbox-config.h']
      dirs: ['bin', 'include', 'lib']

    # Build statistics
    buildstats = [{
        "easybuild-framework_version": "2.7.0",
        "easybuild-easyblocks_version": "2.7.0",
        "timestamp": 1458836674,
        "build_time": 51.85,
        "install_size": 8774526,
        "command_line": ['--configfiles=/apps/etc/easybuild.cfg', '--experimental', '--filter-deps=bzip2,gzip,libreadline,M4,ncurses,zlib', '--installpath=/apps/redhatenterpriseserver-7.1-x86_64', '--module-syntax=Lua', '--modules-tool=Lmod', '--robot=/apps/easybuild/ebfiles:/apps/redhatenterpriseserver-7.1-x86_64/software/EasyBuild/2.7.0-foss-2015a/lib/python2.7/site-packages/easybuild_easyconfigs-2.7.0-py2.7.egg/easybuild/easyconfigs', '--robot-paths=/apps/easybuild/ebfiles:', '--sourcepath=/apps/easybuild/sources', 'linbox.yeb'],
        "modules_tool": ('Lmod', '/usr/local/bin/lmod', '5.9.4'),
        "core_count": 16,
        "cpu_model": "Intel(R) Xeon(R) CPU E5-4617 0 @ 2.90GHz",
        "cpu_speed": 3400.0,
        "cpu_vendor": "Intel",
        "gcc_version": "Using built-in specs.; COLLECT_GCC=gcc; COLLECT_LTO_WRAPPER=/net/nfs4/nfs.hydra/apps/redhatenterpriseserver-7.1-x86_64/software/GCC/4.9.2/bin/../libexec/gcc/x86_64-unknown-linux-gnu/4.9.2/lto-wrapper; Target: x86_64-unknown-linux-gnu; Configured with: ../configure --enable-languages=c,c++,fortran --enable-lto --enable-checking=release --disable-multilib --enable-shared=yes --enable-static=yes --enable-threads=posix --enable-gold=default --enable-plugins --enable-ld --with-plugin-ld=ld.gold --enable-bootstrap --prefix=/apps/redhatenterpriseserver-7.1-x86_64/software/GCC/4.9.2 --with-local-prefix=/apps/redhatenterpriseserver-7.1-x86_64/software/GCC/4.9.2; Thread model: posix; gcc version 4.9.2 (GCC) ; ",
        "glibc_version": "2.17",
        "hostname": "sgi-p30-11.int.s3it.uzh.ch",
        "os_name": "RHEL",
        "os_type": "Linux",
        "os_version": "7.1",
        "platform_name": "x86_64-unknown-linux",
        "python_version": "2.7.5 (default, Apr  9 2015, 11:03:32) ; [GCC 4.8.3 20140911 (Red Hat 4.8.3-9)]",
        "system_gcc_path": "/apps/redhatenterpriseserver-7.1-x86_64/software/GCC/4.9.2/bin/gcc",
        "system_python_path": "/usr/bin/python",
    }]
```

**Comment by riccardomurri** [comment]
For reference, the original `.yeb` file is at <https://github.com/fgeorgatos/easybuild.experimental/blob/master/users/riccardomurri/LinBox-1.4.0-foss-2015a.yeb>

**Comment by boegel** [comment]
fixed in #1822
